packages/swapper/src/swappers/cow/CowApprovalNeeded/CowApprovalNeeded.ts

```diff
@@ -0,0 +1,51 @@
+import { fromAssetId } from '@shapeshiftoss/caip'
+import { SupportedChainIds } from '@shapeshiftoss/types'
+
+import { ApprovalNeededInput, ApprovalNeededOutput, SwapError, SwapErrorTypes } from '../../../api'
+import { erc20AllowanceAbi } from '../../utils/abi/erc20Allowance-abi'
+import { bnOrZero } from '../../utils/bignumber'
+import { getERC20Allowance } from '../../utils/helpers/helpers'
+import { CowSwapperDeps } from '../CowSwapper'
+
+const COW_SWAP_VAULT_RELAYER_ADDRESS = '0xc92e8bdf79f0507f65a392b0ab4667716bfe0110'
+
+export async function CowApprovalNeeded(
+  { adapterManager, web3 }: CowSwapperDeps,
+  { quote, wallet }: ApprovalNeededInput<SupportedChainIds>
+): Promise<ApprovalNeededOutput> {
+  const { sellAsset } = quote
+
+  const { assetReference: sellAssetErc20Address, assetNamespace } = fromAssetId(sellAsset.assetId)
+
+  try {
+    if (assetNamespace !== 'erc20') {
+      throw new SwapError('[CowApprovalNeeded] - unsupported asset namespace', {
+        code: SwapErrorTypes.UNSUPPORTED_NAMESPACE,
+        details: { assetNamespace }
+      })
+    }
+
+    const adapter = await adapterManager.byChainId(sellAsset.chainId)
+    const receiveAddress = await adapter.getAddress({ wallet })
+
+    const allowanceResult = await getERC20Allowance({
+      web3,
+      erc20AllowanceAbi,
+      sellAssetErc20Address,
+      spenderAddress: COW_SWAP_VAULT_RELAYER_ADDRESS,
```

Comment:
just curious, is there an api endpoint we can get this from?

Reply:
I looked but it doesn't seem to exist ...